File: unix/boot/spp/rpp/ratlibf/delete.f
      subroutine delete (symbol, st)
      integer symbol (100)
      integer st
      integer mem( 60000)
      common/cdsmem/mem
      integer stlu
      integer node, pred
      if (.not.(stlu (symbol, node, pred, st) .eq. 1))goto 23000
      mem (pred + 0) = mem (node + 0)
      call dsfree (node)
23000 continue
      return
      end

File: unix/boot/spp/rpp/ratlibr/delete.r
include	defs

# delete --- remove a symbol from the symbol table

   subroutine delete (symbol, st)
   character symbol (ARB)
   pointer st

   DS_DECL(Mem, MEMSIZE)

   integer stlu

   pointer node, pred

   if (stlu (symbol, node, pred, st) == YES) {
      Mem (pred + ST_LINK) = Mem (node + ST_LINK)
      call dsfree (node)
      }

   return
   end
